ddtrace/context.py

```diff
@@ -114,16 +130,19 @@ def _set_current_span(self, span):
             self._parent_span_id = None
 
     def add_span(self, span):
+        # type: (Optional[Span]) -> None
         """
         Add a span to the context trace list, keeping it as the last active span.
         """
         with self._lock:
             self._set_current_span(span)
 
-            self._trace.append(span)
-            span._context = self
+            # TODO: Check if span is None
+            self._trace.append(span)  # type: ignore
```

Comment:
`_set_current_span` can handle `None` as a sort of "null span". Can the append then be skipped  when `span` is `None`?

Reply:
I don't think `span` can/should be `None` here.

Reply:
That's my thinking as well, that's why I added the todo comment for a separate PR (JIRA ticket [here](https://datadoghq.atlassian.net/jira/software/projects/APMPY/boards/136?selectedIssue=APMPY-509))

Reply:
> I don't think `span` can/should be `None` here.

So this should be enforced by the type hint then?